packages/wdio-browserstack-service/src/Percy/PercyBinary.ts

```diff
@@ -0,0 +1,209 @@
+import url from 'node:url'
+import yauzl from 'yauzl'
+import fs from 'node:fs'
+import fsp from 'node:fs/promises'
+import pkg from 'follow-redirects'
+const { https } = pkg
+
+import path from 'node:path'
+import os from 'node:os'
+import { spawn } from 'node:child_process'
+import { PercyLogger } from './PercyLogger.js'
+import type { Options } from '@wdio/types'
+
+class PercyBinary {
+    #hostOS = process.platform
+    #httpPath: any = null
+    #binaryName = 'percy'
+
+    #orderedPaths = [
+        path.join(this.#homedir(), '.browserstack'),
+        process.cwd(),
+        os.tmpdir()
+    ]
+
+    constructor() {
+        const base = 'https://github.com/percy/cli/releases/latest/download'
+        if (this.#hostOS.match(/darwin|mac os/i)) {
+            this.#httpPath = base + '/percy-osx.zip'
+        } else if (this.#hostOS.match(/mswin|msys|mingw|cygwin|bccwin|wince|emc|win32/i)) {
+            this.#httpPath = base + '/percy-win.zip'
+            this.#binaryName = 'percy.exe'
+        } else {
+            this.#httpPath = base + '/percy-linux.zip'
+        }
+    }
+
+    #homedir(): string {
+        if (typeof os.homedir === 'function') {
+            return os.homedir()
+        }
+
+        const env = process.env
+        const home = env.HOME
+        const user = env.LOGNAME || env.USER || env.LNAME || env.USERNAME
+
+        if (process.platform === 'win32') {
+            return ( env.USERPROFILE || (env.HOMEDRIVE || 'null') + env.HOMEPATH || home || null ) as string
+        }
+
+        if (process.platform === 'darwin') {
+            return ( home || (user ? '/Users/' + user : null) ) as string
+        }
+
+        if (process.platform === 'linux') {
+            return ( home || (process.getuid && process.getuid() === 0 ? '/root' : (user ? '/home/' + user : null)) ) as string
+        }
+
+        return (home || null) as string
+    }
+
+    #makePath(path: string) {
+        if (this.#checkPath(path)) {
+            return true
+        }
```

Comment:
use this and remove `#checkPath`
```suggestion
        const hasDir = await fsp.access(path).then(() => true, () => false)
        if (hasDir) {
            return true
        }
```

Reply:
checkPath is used in multiple places, replaced code in checkPath function with suggested code